soccer/gameplay/plays/restarts/out_indirect_pivot.py

```diff
@@ -24,112 +31,181 @@ class State(Enum):
 
         # larger avoid ball radius for line kick setup so we don't run over the ball backwards
-        #self.kicker.setup_ball_avoid = constants.Field.CenterRadius - constants.Robot.Radius
-        #self.kicker.drive_around_dist = constants.Field.CenterRadius - constants.Robot.Radius
+        self.kicker.setup_ball_avoid = constants.Field.CenterRadius - constants.Robot.Radius
+        self.kicker.drive_around_dist = constants.Field.CenterRadius - constants.Robot.Radius
 
         # create a one touch pass behavior with line kick as the skill
-        self.pass_bhvr = tactics.coordinated_pass.CoordinatedPass()
+        self.pass_bhvr = tactics.coordinated_pass.CoordinatedPass(skillkicker=(self.kicker, lambda x: True))
 
         self.add_state(OurIndirectPivot.State.passing,
                        behavior.Behavior.State.running)
         self.add_state(OurIndirectPivot.State.kicking, behavior.Behavior.State.running)
+        self.add_state(OurIndirectPivot.State.preparing, behavior.Behavior.State.running)
 
         # add transistions for when the play is done
         self.add_transition(behavior.Behavior.State.start,
                             OurIndirectPivot.State.passing, lambda: True,
                             'immediately')
 
+        #self.add_transition(OurIndirectPivot.State.preparing,
+        #                    OurIndirectPivot.State.passing, lambda: self.subbehavior_with_name('setupKick').is_done_running(),
+        #                    'move_completed')
+
         self.add_transition(OurIndirectPivot.State.passing,
                             OurIndirectPivot.State.kicking,
-                            lambda: self.pass_bhvr.is_done_running(), 'Pass Completed')
+                            lambda: self.pass_bhvr.is_done_running() or self.ball_has_left, 'Pass Completed')
 
         self.add_transition(OurIndirectPivot.State.kicking,
                             behavior.Behavior.State.completed,
-                            lambda: False, 'Shot Completed')
+                            lambda: time.time() - OurIndirectPivot.LAST_START > OurIndirectPivot.MAX_RUNTIME, 'Shot Completed')
 
     @classmethod
     def score(cls):
         gs = main.game_state()
-
+        if not (gs.is_playing() or gs.is_ready_state()) or (OurIndirectPivot.LAST_START is not None and (time.time() - OurIndirectPivot.LAST_START) > OurIndirectPivot.MAX_RUNTIME):
+            OurIndirectPivot.Running = False
+            OurIndirectPivot.LAST_START = None
+            return float('inf')
         # enter play when doing a corner kick or stay in it even if we manipulate the ball
-        if OurIndirectPivot.Running or (gs.is_ready_state() and (gs.is_our_direct() or gs.is_our_indirect()) and main.ball().pos.y < (
-                constants.Field.Length - 1.2)):
+        if not OurIndirectPivot.Running and OurIndirectPivot.LAST_START is None and (gs.is_ready_state() and gs.is_our_free_kick() and main.ball().pos.y < (
```

Comment:
I can see youre trying to do stuff with score, and it's a great idea, but it might be refactored with our situation analysis. We may keep the evaluation stuff, but we'll probably figure out what we're doing with the scores later